src/components/Marketplace/MarketplaceScreen.js

```diff
@@ -28,55 +34,86 @@ export default function MarketplaceScreen() {
   const [numResults, setNumResults] = useState(10); // # of results to display
 
   const classes = useStyles();
-  // Get records from Airtable whenever DOM mounts and updates/changes
-  useEffect(() => {
+  function getFarmRecords() {
     base('Farms').select({ view: 'Grid view' }).all()
       .then((farmRecords) => {
         setFarmListings(farmRecords);
       });
-    base('Listings').select({ view: 'Grid view' }).all()
+  }
+  // Get prod id, grouped unit type, price per grouped unit for each produce record
+  function getProduceRecords() {
+    // TODO: change Listing UPDATED back to Listings
+    base('Listing UPDATED').select({ view: 'Grid view' }).all()
       .then((produceRecords) => {
-        setProduceListings(produceRecords);
+        const filteredProduceRecords = [];
+        produceRecords.forEach((record) => {
+          const recordInfo = {
+            listingID: record.fields['listing id'],
+            produceID: record.fields.produce ? record.fields.produce[0] : -1,
+            farmID: record.fields['farm id'],
+            unitType: record.fields['grouped produce type'] || 'pallet',
+            unitPrice: record.fields['standard price per grouped produce type'] || -1,
```

Comment:
Remember that we're selling crops by pallets, not units. So to get the price per pallets, we would take the standard price per grouped produce type and grouped produce type per pallet